src/comparator.ml

```diff
@@ -0,0 +1,11 @@
+type 'a t = 'a -> 'a -> Ordering.t
+
+let make (compare: 'a -> 'a -> int) : 'a t = fun this that ->
+  let r = compare this that in if r = 0 then Ordering.Equal
+  else begin
+    if r > 0 then Ordering.Greater
+    else Ordering.Less
+  end 
+
+let makeDefault (this: 'a) (that: 'a): Ordering.t = 
```

Comment:
I still don't understand this one. It doesn't really make anything, even if partially applied. Could it just be `default`?.